airbyte-integrations/bases/base-normalization/normalization/transform_catalog/destination_name_transformer.py

```diff
@@ -0,0 +1,168 @@
+            return True
+        if self.integration_type.value == DestinationType.BIGQUERY.value:
+            return False
+        doesnt_start_with_alphaunderscore = match("[^A-Za-z_]", input_name[0])
+        contains_non_alphanumeric = match(".*[^A-Za-z0-9_].*", input_name)
+        return doesnt_start_with_alphaunderscore or contains_non_alphanumeric
+
+    def normalize_schema_name(self, schema_name: str, in_jinja: bool = False):
+        """
+        @param schema_name is the schema to normalize
+        @param in_jinja is a boolean to specify if the returned normalized will be used inside a jinja macro or not
+        """
+        return self.__normalize_non_column_identifier_name(schema_name, in_jinja)
+
+    def normalize_table_name(self, table_name: str, in_jinja: bool = False):
+        """
+        @param table_name is the table to normalize
+        @param in_jinja is a boolean to specify if the returned normalized will be used inside a jinja macro or not
+        """
+        return self.__normalize_non_column_identifier_name(table_name, in_jinja)
+
+    def normalize_column_name(self, column_name: str, in_jinja: bool = False):
+        """
+        @param column_name is the column to normalize
+        @param in_jinja is a boolean to specify if the returned normalized will be used inside a jinja macro or not
+        """
+        return self.__normalize_identifier_name(column_name, in_jinja)
+
+    # Private methods
+
+    def __normalize_non_column_identifier_name(self, input_name: str, in_jinja: bool = False) -> str:
+        # We force standard naming for non column names (see issue #1785)
+        result = transform_standard_naming(input_name)
+        result = self.__truncate_identifier_name(result)
+        result = self.__normalize_identifier_case(result, is_quoted=False)
+        return result
+
+    def __normalize_identifier_name(self, column_name: str, in_jinja: bool = False):
+        result = self.__truncate_identifier_name(column_name)
+        if self.needs_quotes(result):
+            result = f"adapter.quote('{result}')"
+            result = self.__normalize_identifier_case(result, is_quoted=True)
+            if not in_jinja:
+                result = jinja_call(result)
+            return result
+        else:
+            result = self.__normalize_identifier_case(result, is_quoted=False)
+        if in_jinja:
+            # to refer to columns while already in jinja context, always quote
+            return f"'{result}'"
+        return result
+
+    def __truncate_identifier_name(self, input_name: str) -> str:
+        if self.integration_type.value == DestinationType.BIGQUERY.value:
+            if len(input_name) >= (1024 - TRUNCATE_RESERVED_SIZE):
+                # bigquery has limit of 1024 characters
+                input_name = input_name[0 : (1024 - TRUNCATE_RESERVED_SIZE)]
+            input_name = transform_standard_naming(input_name)
```

Comment:
another question. why is it happening here? The name of the function is about truncation but here you're actually transforming the name

Reply:
You could make the code a lot nicer by having a map of `integration_type` to `size`